### PR TITLE
Fix Cors policy changes may require a site reload to be activated

### LINKED ISSUE #9841: Cors policy changes may require a site reload to be activated
### Describe the bug
Recently adding a new cors policy on a tenant and it just didn't seem to be being acknowledged and the cross domain request was blocked - this is for a web api controller. The I decided to disabled and enable the tenant from the default site and boom it started working.

I am thinking this might require the current tenant to be reloaded for policies to be activated similar to the general settings.

Not sure if anyone else has experienced this.

### To Reproduce
1. Go to 'Settings > CORS'.
2. Click on 'Add policy'.
3. Create a new policy for a cross site request.
4. Make an cross site request that would require the policy - it should not work be cause of missing headers.
5. Disable and renable the tenant and it should now work.

### Expected behavior
Policies should be able to be activated or modified without having to go to the default site to reload the site.

### PR DESCRIPTION
**Pull request by infofromca**
Fix #9841

### CODE REVIEW

**Review by jtkech — APPROVED** [review]

File: src/OrchardCore.Modules/OrchardCore.Cors/Controllers/AdminController.cs [hunk 1]
```diff
@@ -11,12 +11,16 @@
...
+using OrchardCore.Environment.Shell;
 
 namespace OrchardCore.Cors.Controllers
 {
     [Admin]
     public class AdminController : Controller
     {
+        private readonly IShellHost _shellHost;
+        private readonly ShellSettings _shellSettings;
+
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Cors/Controllers/AdminController.cs:23]: Remove extra new line

**Review by hishamco — CHANGES_REQUESTED** [review]
LGTM after removeing the extra line

**Review by hishamco — APPROVED** [review]

**Review by hishamco — CHANGES_REQUESTED** [review]
Seems you miss to add a warning alter in the view like what usually did in other modules

https://github.com/OrchardCMS/OrchardCore/blob/c255456bf1d9093d13447e17485a23a1253f3e9b/src/OrchardCore.Modules/OrchardCore.Localization/Views/LocalizationSettings.Edit.cshtml#L5

**Comment by infofromca** [comment]
just add a warning alter. by following OpenId module.
by the way, I found that there is also no  warning alter on the Recipe module

File: src/OrchardCore.Modules/OrchardCore.Cors/Views/Admin/Index.cshtml [hunk 2]
```diff
@@ -146,7 +146,7 @@
             </div>
         </div>
         <div class="mb-3">
-            <button type="button" class="btn btn-primary" v-on:click="$emit('ok', policy, $event)">@T["Save"]</button>
+            <button type="button" class="btn btn-primary" v-on:click="$emit('ok', policy, $event)">@T["Save and reload the tenant"]</button>
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Cors/Views/Admin/Index.cshtml:149]: `Save` is fine, but where's the alert?

**Comment by hishamco** [comment]
> by the way, I found that there is also no warning alter on the Recipe module

Let me verify and fix if there's something missing

**Comment by hishamco** [comment]
The PR #12069 is created

**Comment by rserj** [comment]
@sebastienros can we merge it? 
@jtkech I have multiple instances of Orchard Core running at the same time, will Tenant restart on all of the stacks?

**Comment by hishamco** [comment]
@rserj I requested few changes that are not applied yet!!

**Comment by jtkech** [comment]
@rserj 

Yes if you have a stateless config e.g. shell settings coming from database and enabled the Tenant.Distributed feature needing e.g a concrete implemenation of a distributed cache e.g. Redis cache and then recommended Redis lock.

**Comment by hishamco** [comment]
@infofromca could apply the requested changes?

**Review by hishamco — APPROVED** [review]

**Pull request merged**